### PR TITLE
docs: add installation and build guides from wiki to repo

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,485 @@
+- **IMPORTANT**: Before upgrading to a new version, **always check for [breaking changes](https://neovim.io/doc/user/news.html#news-breaking).**
+- 👀  See Table of Contents by expanding the _Pages_ sidebar to the right ➡️
```
Comment (clason) [inline BUILD.md:2]:
```suggestion
```

**Comment by wookayin** [comment]
How about having a top-level `docs/` folder?

**Comment by clason** [comment]
Yes, these two files are special since they are mainly useful _before_ you have built and/or installed Neovim. For everything else, the goal is to move that into actual (vimdoc) help documents eventually (after pruning/updating), or offload to neovim.io.

(Also, welcome to the team! 🥳 )

**Comment by clason** [comment]
Possible follow-ups (note for future reference):
* move https://github.com/neovim/neovim/wiki/FAQ to `faq.txt`
* move https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim (`dev-vim-patches.txt`? or merge with `MAINTAIN.md`/`CONTRIBUTING.md`?)

**Comment by dundargoc** [comment]
I like dev-vim-patches more.

**Comment by dundargoc** [comment]
@justinmk review

**Comment by justinmk** [comment]
> these two files are special since they are mainly useful _before_ you have built and/or installed Neovim

Can we also put these in runtime/doc/ :) Because the website needs an "Install" page, and might as well also give it a "Build" page. The README can link to the `runtime/doc/*.txt` files using relative paths so there isn't really a discoverability issue there.

Of course we could have a special-purpose "Install" page on the website, but I'd rather focus on perfecting one page instead of multiple pages. It also drives improvements to :help, such as URL rendering, images, etc.

**Comment by clason** [comment]
They're quite heavy on the links, though, which would make them significantly harder to navigate after pushing them through the gen_html_help sausage grinder...

`INSTALLING` and `BUILDING` are also typical files in a source repo, so much more visible to people coming here.

So I'd prefer to leave them here, at least for now; this round of cleaning is not the last round -- after we've cleaned out the Wiki, the next step will be consolidating (and moving) all the various markdown files.

**Comment by justinmk** [comment]
Fair enough.

**Comment by dundargoc** [comment]
Personally I don't mind either way, except that I want this done speedily so the information isn't fragmented in two repos for long. Let's move forward with this for now so we can clean up the wiki and we can deliberate more on how to best present this information later.

**Review by clason — APPROVED** [review]

**Pull request merged**